### PR TITLE
Autotools macros update

### PR DESCRIPTION
**Pull request by daveol**
A lot of macro's where obsolete look [here](http://www.gnu.org/software/autoconf/manual/autoconf-2.62/html_node/Obsolete-Macros.html) for more info

### CODE REVIEW

File: config/stl_hash.m4 [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copied from the protobuf package, http://code.google.com/p/protobuf/
+	# Copied from the protobuf package, http://code.google.com/p/protobuf/
```
Comment (peternewman) [inline config/stl_hash.m4]: Remove these extra spaces, we should probably update their URL too.

**Comment by nomis52** [comment]
We need to be careful with this, not everyone has the latest autoconf. Unfortunately the page linked to doesn't tell us when the new macros became available.

Looking at releases, precise is the oldest I know of with 2.68. Luckily this is what travis runs!

**Comment by nomis52** [comment]
That said, LGTM.

File: config/stl_hash.m4 (hunk 1 above)
Comment (daveol) [inline config/stl_hash.m4]: I don't even know where those spaces came from... I will update the url but I updated the macro's myself

**Comment by daveol** [comment]
CentOS6 (a.k.a. el6) is also on autoconf 2.68, and debian squeeze (oldoldstabe a.k.a. 2 versions behind) has 2.67 so I think we are safe

**Comment by nomis52** [comment]
LGTM

**Comment by peternewman** [comment]
FWIW, our limitation was actually 'AM_PROG_AR' which was introduced in Automake 1.11.2. Ubuntu 12.04 LTS has it, but 10.04 (EOL April this year) didn't have it. We weren't actually specifying a minimum version of automake, so I've added that in https://github.com/OpenLightingProject/ola/pull/981 .

**Comment by peternewman** [comment]
Further info, @daveol and @nomis52 the Automake 1.11.2/AM_PROG_AR thing breaks mingw32. It looks like we should actually be moving to mingw-w64, which despite the name supports 32 and 64 bit. However, in the short term I wonder if we should revert that particular bit of this change, until we've got mingw-w64 working.

**Comment by nomis52** [comment]
Can you paste links or log output please. 

Do you know if mingw-w64 works for us??

**Comment by peternewman** [comment]
Finally as promised:

autoreconf -i
configure.ac:30: warning: macro `AM_PROG_AR' not found in library
configure.ac:30: warning: macro `AM_PROG_AR' not found in library
configure.ac:30: error: possibly undefined macro: AM_PROG_AR
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
autoreconf-2.68: /mingw/bin/autoconf-2.68 failed with exit status: 1

We've discussed on IRC @daveol and @nomis52 agreed we'd revert this change for 0.10, then move to mingw-w64 for 0.11. Do you know if it's just that one line AM_PROG_AR we need to revert @daveol it seems to build fine with the other changes still present, but it would be good to know if the LT_INIT win32-dll stuff is related, or just a parallel tidy?

**Comment by daveol** [comment]
Probably only the AM_PROG_AR since the LT_INIT is a separate thing the only thing related is that libtool (a.k.a. LT_INIT ,a.k.a (deprecated) AC_PROG_LIBTOOL) uses ar (a.k.a. AM_PROG_AR)

**Comment by peternewman** [comment]
Okay thanks @daveol . Out of interest, did you have Doxygen on the box? When I run with -Wall -Werror I get lots of Doxygen errors too.

**Comment by daveol** [comment]
I had doxygen, but I couldn't figure out how to fix those errors

**Comment by peternewman** [comment]
Ah okay @daveol. Done in #998 anyway.

**Pull request merged**